genienlp/tasks/almond/__init__.py

```diff
@@ -31,86 +31,83 @@
 import torch
 import logging
 from tqdm import tqdm
+from collections import defaultdict
 
 from ..base_task import BaseTask
 from ..registry import register_task
-from .. import generic_dataset
+from ..generic_dataset import CQA
 from ...data_utils.example import Example
 
+from ..base_dataset import Split
+
 logger = logging.getLogger(__name__)
 
 
-class AlmondDataset(generic_dataset.CQA):
+class AlmondDataset(CQA):
     """Obtaining dataset for Almond semantic parsing task"""
 
     base_url = None
 
-    def __init__(self, path, *, make_example, subsample=None, cached_path=None, skip_cache=False,
-                 **kwargs):
+
+    def __init__(self, path, *, make_example, subsample=None, cached_path=None, skip_cache=False, examples=None, **kwargs):
+        
+        if examples:
+            super().__init__(examples, **kwargs)
+            return
+            
         cache_name = os.path.join(cached_path, os.path.basename(path), str(subsample))
 
         if os.path.exists(cache_name) and not skip_cache:
             logger.info(f'Loading cached data from {cache_name}')
             examples = torch.load(cache_name)
         else:
             examples = []
-
             n = 0
             with open(path, 'r', encoding='utf-8') as fp:
                 for line in fp:
                     n += 1
 
             max_examples = min(n, subsample) if subsample is not None else n
             for i, line in tqdm(enumerate(open(path, 'r', encoding='utf-8')), total=max_examples):
-                if i >= max_examples:
-                    break
-
                 parts = line.strip().split('\t')
-                examples.append(make_example(parts))
+                examples.append(make_example(parts, language=kwargs.get('language', None)))
                 if len(examples) >= max_examples:
                     break
             os.makedirs(os.path.dirname(cache_name), exist_ok=True)
             logger.info(f'Caching data to {cache_name}')
             torch.save(examples, cache_name)
 
         super().__init__(examples, **kwargs)
+        
 
     @classmethod
-    def splits(cls, root='.data', task_name='almond', train='train', validation='eval', test='test', **kwargs):
+    def return_splits(cls, path, train='train', validation='eval', test='test', **kwargs):
```

Comment:
Why are you renaming this method and not the one in the other classes?

Reply:
because I am using 'path' instead of 'root' and pylint complains that method does not match with Dataset class.